pkg/auth/rights/rights.go

```diff
@@ -17,106 +17,132 @@ package rights
 
 import (
 	"context"
+	"sync"
 
 	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
 )
 
+// RightsMap ...
+type RightsMap struct {
+	syncMap *sync.Map
```

Comment:
Instead of making `syncMap` a pointer, make `NewContext` and `FromContext` carry the `*Rights`

Reply:
Fixed https://github.com/TheThingsNetwork/lorawan-stack/pull/5887/commits/a9f7db7b915b193cddabf450fd57c84a2d05e886